src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/DefaultOpenDistroSecurityKeyStore.java

```diff
@@ -290,71 +293,77 @@ private void initSSLConfig() {
                         /*
                          * for (int i = 0; i < transportKeystoreCert.length; i++) { X509Certificate
                          * x509Certificate = transportKeystoreCert[i];
-                         * 
+                         *
                          * if(x509Certificate != null) {
                          * log.info("Transport keystore subject DN no. {} {}",i,x509Certificate.
                          * getSubjectX500Principal()); } }
                          */
                     } else {
                         throw new ElasticsearchException(
-                                "No certificates found in " + keystoreFilePath + " with alias " + keystoreAlias);
+                            "No certificates found in " + keystoreFilePath + " with alias " + keystoreAlias);
                     }
 
                     final KeyStore ts = KeyStore.getInstance(truststoreType);
                     ts.load(new FileInputStream(new File(truststoreFilePath)),
-                            (truststorePassword == null || truststorePassword.length() == 0) ? null
-                                    : truststorePassword.toCharArray());
+                        (truststorePassword == null || truststorePassword.length() == 0) ? null
+                            : truststorePassword.toCharArray());
 
                     final X509Certificate[] trustedTransportCertificates = SSLCertificateHelper
-                            .exportRootCertificates(ts, truststoreAlias);
+                        .exportRootCertificates(ts, truststoreAlias);
+
 
                     if (trustedTransportCertificates == null) {
                         throw new ElasticsearchException("No truststore configured for server");
                     }
 
+                    validateNewTransportCerts(transportKeystoreCert);
```

Comment:
is it possible to move line 319 to 326 into another function and reuse that code for 344 to avoid duplicity ? You need to handle password requirement with conditions.

Reply:
We should also see if we can remove duplicity between initTransportCert() and initHTTPCert()

Reply:
Sure. I created an issue in our backlog for the same https://github.com/opendistro-for-elasticsearch/security/issues/254